172.md

```diff
@@ -0,0 +1,94 @@
+NIP-172
+=======
+
+Moderated Communities (Reddit Style)
+------------------------------------
+
+`draft` `optional` `author:vitorpamplona`
+
+The goal of this NIP is to create moderator-approved public communities around a topic. It defines the replaceable event `34550` to define the community and the current list of moderators/administrators. Users that want to post into the community, simply tag any Nostr event with an `a` tag. Moderators issue an approval event `34551` that links the community with the new post.
+
+# Community definition
+
+Kind 34550 should include any field that helps define the community and the set of moderators.
+
+```js
+{
+  "id": "<32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>",
+  "pubkey": "<32-bytes lowercase hex-encoded public key of the event creator>",
+  "created_at": "<Unix timestamp in seconds>",
+  "kind": "34550",
+  "tags": [
+    ["d", "<community_name>"],
+    ["description", "<community_description>"],
+	["image",  "<communityimagen>", "WidthxHeight"]
+	
+	//.. other tags relevant to defining the community
+
+    // moderators
+    ["p", "<32-bytes hex of a pubkey>", "<optional recommended relay URL>", "moderator"],
+    ["p", "<32-bytes hex of a pubkey>", "<optional recommended relay URL>", "moderator"],
+    ["p", "<32-bytes hex of a pubkey>", "<optional recommended relay URL>", "moderator"],
+	
+	// relays used by the community 
+    ["relay", "<relay hosting author kind 0>", "author"],
+    ["relay", "<relay where to post requests to and fetch approvals from>"], 
+    ["relay", "<relay where to post requests to and fetch approvals from>"]
+  ]
+}
+```
+
+# New Post Request
+
+Any Nostr event can be a post request. Clients should simply add the community's `a` tag to be presented for the moderator's approval.
+
+```js
+{
+  "id": "<32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>",
+  "pubkey": "<32-bytes lowercase hex-encoded public key of the event creator>",
+  "created_at": "<Unix timestamp in seconds>",
+  "kind": "1",
+  "tags": [
+    ["a", "34550:<community event author pubkey>:<d-identifier of the community>", "<optional relay url>"],
+  ], 
+  "content": "<my content>"
+}
+```
+
+Community management clients can filter all mentions of the kind-34550 event and request moderators to approve each submission. The same moderator can remove his/her approval of the post at any time.
+
+# Post Approval by moderators
+
+The post-approval event includes a stringified `new post request` event inside the `.content` of the approval (NIP-18-style).
+
+```js
+{
+  "id": "<32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>",
+  "pubkey": "<32-bytes lowercase hex-encoded public key of the event creator>",
+  "created_at": "<Unix timestamp in seconds>",
+  "kind": "34551",
```

Comment:
This parameterized replaceable event is missing a "d" tag. Considering the user can be a moderator on many communities and may approve the same event id for more than one group, i think it would be this giant string: `["d", "34550:<community event author pubkey>:<d-identifier of the community>:<Post Request ID>"]`.

Maybe it is better as a regular event in the `1000 <= n < 10000` range like `4550` or  `4551` with no "d" tag. I don't know.

Reply:
Good point. I think we can move it to a regular event with event deletion to revert things.